website/blog/2024-05-24-Agent/index.mdx

```diff
@@ -0,0 +1,142 @@
+agents still perceive it as a single entity to communicate to.
+
+With this definition, we can incorporate both very simple agents that can solve simple tasks with
+a single backend, but also we can have agents that are composed of multiple simpler agents. One can recursively build up more powerful agents. The
+agent concept in AutoGen can cover all these different complexities.
+
+## What are pros and cons of multi vs. single agent?
+
+This question is asked in a variety of ways.
+> Why shall I use multiple agents instead of a single agent?
+
+> Why think about multi agents when we don't have a strong single agent?
+
+> Does multi-agent increase the complexity, latency and cost?
+
+> When shall I use multi-agents vs. single agent?
+
+When we use the word multi-agent and single-agent, I think there
+are at least two different dimensions we need to think about.
+
+* Interface. That means from the user's point of view, do they interact with the system in a single
+interaction point or do they see explicitly multiple agents working and need to interact with multiple of
+them?
+* Architecture. Are there multiple agents underneath running at the backend?
+
+A particular system can have a single-agent interface and a multi-agent architecture, but the users don't need to know that.
+
+### Interface
+
+In some application it's better to have a single interaction point for
+the user to simplify it. But in other applications, for example, when the application is about having multiple
+agents debate about a subject, the users need to see what each agent says.
+In that case it's beneficial for them to actually see the multi agents' behavior. Another example is the social
+simulation experiment: People also want to see the behavior of each agent.
+
+### Architecture
+
+The multi-agent design of the architecture is easier to maintain,
+understand and extend than a single agent system. Even for the single agent
+based interface, a multi-agent implementation can potentially make the system more modular, and
+easier for developers to add or remove components of functionality. It's very important to
+recognize that the multi-agent architecture is a good way to build a single agent. While not obvious, it has a root in the society of mind theory by Marvin Minsky in 1986.
+One can start from relatively simple agents, compose them together and make them interact
+in a good way. Then they would exhibit a higher level of intelligence.
+
+We don't have a good single agent that
+can do everything we want yet. And why is that? It could be because we haven't figured out the right
+way of composing the multi-agent to build this powerful single agent. But firstly, we need to have the
+framework that allows easy experimentation of these different ways of combining models and agents. My own experience is that if people practise using multi agents to solve a problem, they often reach a solution faster.
```

Comment:
This is a bit unclear. We can provide some examples of compositions, e.g. group chat turned out to be a simple yet powerful composition. Others like graph based group chat also.